### PR TITLE
PS-6057: Suppress gcc and clang warnings for RocksDB and TokuDB

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

**Comment by inikep** [comment]
Tested at https://inikep.visualstudio.com/mysql-server/_build/results?buildId=86

**Comment by inikep** [comment]
Linking issues for clang-7 and clang-8 are:
```
/usr/bin/ld.gold: error: /usr/lib/llvm-8/lib/clang/8.0.1/lib/linux/libclang_rt.fuzzer-x86_64.a(FuzzerDataFlowTrace.cpp.o): invalid section group 9 refers to earlier section 3
/usr/bin/ld.gold: error: /usr/lib/llvm-8/lib/clang/8.0.1/lib/linux/libclang_rt.fuzzer-x86_64.a(FuzzerDataFlowTrace.cpp.o): invalid section group 10 refers to earlier section 4
/usr/bin/ld.gold: error: /usr/lib/llvm-8/lib/clang/8.0.1/lib/linux/libclang_rt.fuzzer-x86_64.a(FuzzerDataFlowTrace.cpp.o): invalid section group 11 refers to earlier section 5
/usr/bin/ld.gold: error: /usr/lib/llvm-8/lib/clang/8.0.1/lib/linux/libclang_rt.fuzzer-x86_64.a(FuzzerDriver.cpp.o): invalid section group 63 refers to earlier section 6
/usr/bin/ld.gold: error: /usr/lib/llvm-8/lib/clang/8.0.1/lib/linux/libclang_rt.fuzzer-x86_64.a(FuzzerDriver.cpp.o): invalid section group 64 refers to earlier section 8
/usr/bin/ld.gold: error: /usr/lib/llvm-8/lib/clang/8.0.1/lib/linux/libclang_rt.fuzzer-x86_64.a(FuzzerDriver.cpp.o): invalid section group 65 refers to earlier section 9
```

File: plugin/tokudb-backup-plugin/CMakeLists.txt [hunk 1]
```diff
@@ -26,23 +26,19 @@ IF(NOT TOKUDB_OK OR NOT HAVE_O_DIRECT OR NOT HAVE_CLOCK_MONOTONIC OR APPLE)
...
-append_cflags_if_supported(-Wno-vla)
+
+
+# Suppress warnings for all compilers
+remove_compile_flags(-Wvla)
+
+
+# Suppress warnings for all clang versions
+IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+append_cflags_if_supported(-Wno-deprecated-dynamic-exception-spec)
```
Comment (percona-ysorokin) [inline plugin/tokudb-backup-plugin/CMakeLists.txt]: indentation

File: storage/rocksdb/CMakeLists.txt [hunk 2]
```diff
@@ -27,41 +27,68 @@ IF (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
...
-remove_compile_flags(-Wundef)
+MY_CHECK_CXX_COMPILER_FLAG("-Wno-shadow-field" HAVE_NO_SHADOW_FIELD)
+IF(HAVE_NO_SHADOW_FIELD)
+  ADD_COMPILE_FLAGS(ha_rocksdb.cc rocksdb/utilities/geodb/geodb_impl.cc rocksdb/memtable/hash_skiplist_rep.cc rocksdb/memtable/hash_cuckoo_rep.cc COMPILE_FLAGS "-Wno-shadow-field")
+ENDIF()
+
+
+# Suppress warnings for all clang versions
+IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+remove_compile_flags(-Winconsistent-missing-destructor-override -Wdeprecated -Wextra-semi -Wnon-virtual-dtor)
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt]: Indentation

File: storage/tokudb/CMakeLists.txt [hunk 3]
```diff
@@ -78,17 +78,33 @@ IF(DEFINED TOKUDB_NOPATCH_CONFIG)
...
-    (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 9.0 OR
-     CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0))
-  append_cflags_if_supported(-Wno-address-of-packed-member
-                             -Wno-deprecated-copy)
+remove_compile_flags(-Wvla)
+
+
+# Suppress warnings for all clang versions
+IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+append_cflags_if_supported(-Wno-deprecated-dynamic-exception-spec -Wno-inconsistent-missing-override)
```
Comment (percona-ysorokin) [inline storage/tokudb/CMakeLists.txt]: Indentation

**Review by percona-ysorokin — APPROVED** [review]
LGTM with minor formatting comments

**Pull request merged**